dll/win32/browseui/shellbrowser.cpp

```diff
@@ -3619,15 +3624,15 @@ LRESULT CShellBrowser::OnCreate(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &b
 LRESULT CShellBrowser::OnDestroy(UINT uMsg, WPARAM wParam, LPARAM lParam, BOOL &bHandled)
 {
     HRESULT hr;
+    SaveViewState();
 
     /* The current thread is about to go down so render any IDataObject that may be left in the clipboard */
     OleFlushClipboard();
 
     // TODO: rip down everything
     {
+        m_Destroyed = true; // Ignore browse requests from Explorer band TreeView during destruction
```

Comment:
Is it possible to move this flag set before the call to SaveViewState, for example?

Reply:
In its current state, yes but if `m_Destroyed`  starts to be checked in other places, it might block saving the state and we do want to save the state when the user closes the window.

At the start of `OnDestroy`, everything is still alive (all child windows are valid and have not started destruction). We can however not allow new windows to be created as a child of the shell browser (windows are created during navigation).